Source/WebCore/rendering/ReferencedSVGResources.cpp

```diff
@@ -60,8 +60,39 @@ WTF_MAKE_ISO_ALLOCATED_IMPL(CSSSVGResourceElementClient);
 
 void CSSSVGResourceElementClient::resourceChanged(SVGElement&)
 {
-    if (!m_clientRenderer.renderTreeBeingDestroyed())
-        m_clientRenderer.repaint();
+    if (m_clientRenderer.renderTreeBeingDestroyed())
+        return;
+
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+    auto useUpdateLayerPositionsLogic = [&]() -> std::optional<CheckedPtr<RenderLayer>> {
+        auto& document = m_clientRenderer.document();
+        if (!document.settings().layerBasedSVGEngineEnabled())
+            return std::nullopt;
+
+        // Don't attempt to update anything during layout - the post-layout phase will invoke RenderLayer::updateLayerPosition(), if necessary.
+        if (document.view()->layoutContext().isInLayout())
```

Comment:
This is new: make sure we only use the `updateLayerPositions()` logic if it's safe to do so: not during layout in general, and not if the `m_clientRenderer` needs layout().

During layout repainting of old/new boundaries is already handled using `LayoutRepainter` - no need to handle it here (and would be wrong to do so, leading to debug assertions).